nx-dev/feature-search/src/lib/algolia-search.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useCallback, useRef, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import Link from 'next/link';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import { DocSearchModal, useDocSearchKeyboardEvents } from '@docsearch/react';
+
+declare var document: any;
+declare var navigator: any;
+declare var window: any;
+
+const ACTION_KEY_DEFAULT = ['Ctrl ', 'Control'];
+const ACTION_KEY_APPLE = ['⌘', 'Command'];
+
+function Hit({ hit, children }) {
+  return (
+    <Link href={hit.url}>
+      <a>{children}</a>
+    </Link>
+  );
+}
+
+export function AlgoliaSearch() {
+  const frameworkFilter = `framework:react`; // TODO: @ben Tap into current framework selection
+  const versionFilter = `version:latest`; // TODO: @ben Tap into current version selection
+
+  const router = useRouter();
+  const [isOpen, setIsOpen] = useState(false);
+  const searchButtonRef = useRef();
+  const [initialQuery, setInitialQuery] = useState(null);
+  const [browserDetected, setBrowserDetected] = useState(false);
+  const [actionKey, setActionKey] = useState(ACTION_KEY_DEFAULT);
+
+  const onOpen = useCallback(() => {
```

Comment:
Minor nitpick... usually in React-world devs used `handleX` for event handlers rather than `onX`. So `handleOpen`, `handleClose` and `handleInput` in these cases.